lib/DebugInfo/Symbolize/BugsnagSymbolize.cpp

```diff
@@ -0,0 +1,100 @@
+#include "llvm/DebugInfo/Symbolize/BugsnagSymbolize.h"
+#include "llvm/DebugInfo/Symbolize/Symbolize.h"
+#include <cstdlib>
+#include <cstring>
+
+using namespace llvm;
+
+static void freeAndInvalidate(void** p) {
+  free((void*)*p);
+  *p = nullptr;
+}
+
+static void destroySymbolizeResult(SymbolizeResult* symbolizeResult) {
+  if (!symbolizeResult) {
+    return;
+  }
+  freeAndInvalidate((void**)&symbolizeResult->fileName);
+  freeAndInvalidate((void**)&symbolizeResult->shortFunctionName);
+  freeAndInvalidate((void**)&symbolizeResult->linkageFunctionName);
+  freeAndInvalidate((void**)&symbolizeResult->symbolTableFunctionName);
+}
+
+static void destroySymbolizeResults(SymbolizeResults* symbolizeResults) {
+  if (!symbolizeResults) {
+    return;
+  }
+  for (int i = 0; i < symbolizeResults->resultCount; i++) {
+    destroySymbolizeResult(&symbolizeResults->results[i]);
+  }
+}
+
+SymbolizeResult getSymbolizeResult(const DILineInfo &info, int64_t address, bool inlined) {
+  SymbolizeResult result = {0};
+
+  result.address = address;
+  result.inlined = inlined;
+
+  unsigned int len = std::strlen(info.FileName.c_str());
```

Comment:
strlen returns the length of the string not including the terminating null character (https://www.cplusplus.com/reference/cstring/strlen/), however I believe strcpy will copy the string including the terminating null character (https://www.cplusplus.com/reference/cstring/strcpy/) - so I think this design pattern is causing an off by 1 buffer overflow.

Reply:
Updated to account for null terminating character